src/lazy/expanded/macro_evaluator.rs

```diff
@@ -1091,6 +1202,25 @@ mod tests {
         )
     }
 
+    #[test]
+    fn annotate() -> IonResult<()> {
```

Comment:
Might be a good idea to have another test case where you invoke `annotate` as an e-expression rather than as a TDL macro invocation. (Just in case there is a regression, it would (hopefully) make it easier to isolate the problem.)

Reply:
Tracking in #808.